justspacesapp/urls.py

```diff
@@ -13,9 +13,98 @@
     1. Import the include() function: from django.urls import include, path
     2. Add a URL to urlpatterns:  path('blog/', include('blog.urls'))
 """
-from django.conf.urls import url, include
-from justspacesapp.views import CreateNewSurvey
+from django.conf.urls import url
+from django.utils.translation import ugettext_lazy as _
+from justspacesapp.views import AgencyCreate, LocationCreate, StudyCreate, \
+                                SurveyListEdit, SurveyListView
+import fobi.views
 
 urlpatterns = [
-    url(r'^$', CreateNewSurvey.as_view(), name='create-new-survey'),
+    url(r'^$',
+        SurveyListView.as_view(),
+        name='home'),
+
+    url(r'agencies/create$',
+        AgencyCreate.as_view(),
+        name='agencies-create'),
+
+    url(r'locations/create$',
+        LocationCreate.as_view(),
+        name='locations-create'),
+
+    url(r'studies/create$',
+        StudyCreate.as_view(),
+        name='studies-create'),
+
+    # Create new survey
+    url(r'surveys/create$',
+        view=fobi.views.create_form_entry,
+        name='fobi.create_form_entry'),
+
+    # Create survey element
+    url(_(r'^surveys/elements/create/(?P<form_entry_id>\d+)/'
+          r'(?P<form_element_plugin_uid>[\w_\-]+)/$'),
+        view=fobi.views.add_form_element_entry,
+        name='fobi.add_form_element_entry'),
+
+    # Edit survey element
+    url(_(r'^surveys/elements/edit/(?P<form_element_entry_id>\d+)/$'),
+        view=fobi.views.edit_form_element_entry,
+        name='fobi.edit_form_element_entry'),
+
+    # Delete survey element
+    url(_(r'^surveys/elements/delete/(?P<form_element_entry_id>\d+)/$'),
+        view=fobi.views.delete_form_element_entry,
+        name='fobi.delete_form_element_entry'),
+
+    # Create survey handler
+    url(_(r'^surveys/handlers/create/(?P<form_entry_id>\d+)/'
+          r'(?P<form_handler_plugin_uid>[\w_\-]+)/$'),
+        view=fobi.views.add_form_handler_entry,
+        name='fobi.add_form_handler_entry'),
+
+    # Edit survey handler
+    url(_(r'^surveys/handlers/edit/(?P<form_handler_entry_id>\d+)/$'),
+        view=fobi.views.edit_form_handler_entry,
+        name='fobi.edit_form_handler_entry'),
+
+    # Delete survey handler
+    url(_(r'^surveys/handlers/delete/(?P<form_handler_entry_id>\d+)/$'),
+        view=fobi.views.delete_form_handler_entry,
+        name='fobi.delete_form_handler_entry'),
+
+    # Edit survey
+    url(_(r'^surveys/edit/(?P<form_entry_id>\d+)/$'),
+        view=fobi.views.edit_form_entry,
+        name='fobi.edit_form_entry'),
+
+    # Delete survey
+    url(_(r'^surveys/delete/(?P<form_entry_id>\d+)/$'),
+        view=fobi.views.delete_form_entry,
+        name='delete_form_entry'),
+
+    # Survey list edit
```

Comment:
We want to use query params to distinguish between edit and run functionality.  That's the main point of this issue: https://github.com/datamade/just-spaces/issues/26

Reply:
Ah, I see. Combining into one list for now since we don't have published or unpublished states attached to surveys yet.